### PR TITLE
Backports for 4.0.2

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Backport to 4.0 of Taipy GUI issues
#2212 ([PR#2250](https://github.com/Avaiga/taipy/pull/2256)),
#2242 ([PR#2250](https://github.com/Avaiga/taipy/pull/2250)),
#2281 ([PR#2315](https://github.com/Avaiga/taipy/pull/2315)),
#2286 ([PR#2300](https://github.com/Avaiga/taipy/pull/2300)),
#2302 ([PR#2314](https://github.com/Avaiga/taipy/pull/2314)), and
#2305 ([PR#2313](https://github.com/Avaiga/taipy/pull/2313)).

### CODE REVIEW

File: frontend/taipy-gui/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "taipy-gui",
-  "version": "4.0.1",
+  "version": "4.0.2",
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/package.json:3]: there are more package.json files to change I believe

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx [hunk 2]
```diff
@@ -279,6 +280,11 @@ const updateArrays = (sel: number[][], val: number[], idx: number) => {
     return sel;
 };
 
+const getDataKey = (columns: Record<string, ColumnDesc>, decimators?: string[]): [string[], string] => {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Chart.tsx]: I think there's an issue with columns being undefined here
I fixed it locally but did not push it to develop yet

**Review by FredLL-Avaiga — DISMISSED** [review]

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx]:
```py
const getDataKey = (columns?: Record<string, ColumnDesc>, decimators?: string[]): [string[], string] => {
    const backCols = columns ? Object.values(columns).map((col) => col.dfid) : [];
    return [backCols, backCols.join("-") + (decimators ? `--${decimators.join("")}` : "")];
};
```

File: frontend/taipy-gui/package.json (hunk 1 above)
Comment (FabienLelaquais) [reply frontend/taipy-gui/package.json:3]: 👍

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx (hunk 2 above)
Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx]: In there.

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Chart.tsx]: we need this

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

**Review by FredLL-Avaiga — APPROVED** [review]
👍

**Pull request merged**